warehouse/accounts/views.py

```diff
@@ -53,6 +53,7 @@ def failed_logins(exc, request):
 @view_config(
     route_name="accounts.profile",
     renderer="accounts/profile.html",
+    uses_session=True,
```

Comment:
You'd mix it in with javascript instead, it's something that's easy enough to fail gracefully back if that fails for whatever reason. I think to do it really nicely would require a bit of work on the ``html_include()`` functionality so that it can have a default value that gets replaced with the fetched data.

So you'd end up with template code like:

```
{% htmlinclude request.route_path("includes.user-uploaded-message") %}
<p>{{ user.name|default(user.username, true) }} has not uploaded any projects to PyPI, yet.</p>
{% endhtmlinclude %}
```

Which would initially render the bit that's inside the blocks, and would fetch on the client side using JS the URL given and replace the block with the results of that.

Please note this will NOT currently work, but if that sounds reasonable I could work up the functionality of the current ``html_include()`` into something like that, which would work then.